sled-agent/src/support_bundle/storage.rs

```diff
@@ -0,0 +1,1393 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
This is the meat-and-potatoes of this PR -- I tried to add quite a few tests here. Happy to add more, as requested!